guides/dsl.md

```diff
@@ -0,0 +1,289 @@
+    return [
+      // bind a record tagged "greeting" with an attribute "message"
+      record("greeting", {message: `Hello ${person.name}`})
+    ];
+  });
+```
+
+The equivalent Eve block would be
+
+```eve
+search
+  person = [#person]
+
+bind
+  [#greeting message: "Hello {{person.name}}"]
+```
+
+The identity of a `record()` is established by everything within the parentheses. Therefore, the record `record("greeting", {message: `Hello ${person.name}`})` will create one record for every `person.name`.
+
+If would like to commit records instead of binding them, you can use the `commit()` method in place of `bind()`. The implication here is that you can no longer create blocks which both commit and bind records at the same time.
+
+The DSL supports methods which mirror a couple of the Eve update operators. They are:
+
+### add()
+
+ Adds attribute/value pairs to a record. These added values do not contribute to the identity of a record, allowing you to add multiple values to a single record. This is equivalent to the Eve `+=` operator. For example:
+
+```javascript
+program.bind("Invite classmates to my party", ({find, record}) => {
+  // search for records tagged "friend"
+  let student = find("student");
+  return [
+    // bind a single record tagged "guest-list", with every student as a guest
+    record("guest-list").add({guest: student})
+  ];
+});
+```
+
+### remove()
+
+Removes attribute/value pairs from a record. This is equivalent to the Eve `-=` operator. Remove should only be used in the context of a `commit` block. For example:
+
+```javascript
+program.commit("blacklisted people cannot come to the party", ({find, record}) => {
+  let blacklisted = find("blacklisted");
+  let guests = find("guest-list")
+  return [
+    // remove anyone blacklisted from the guest list
+    guests.remove({guest: blacklisted})
+  ];
+});
+```
+
+You can also use `remove()` to completely delete records or attributes on a record. This is equivalent to `:= none` in Eve. For example:
+
+```javascript
+   // completely remove the target record
+   record.remove() 
+   // remove the attribute on the target record
+   record.remove("attribute")
+```
+
+The DSL does not contain an equivalent for the Eve set operator `:=`, but you can mimic its behavior by chaining the `add()` and `remove()` operators. For example:
+
+```javascript
+program.bind("Reassign Artemis' teacher", ({find, record}) => {
+  let artemis = find("student", {name: "Artemis"});
+  let teacher = find("teacher", {name: "Smith"});
+  return [
+    artemis.remove("teacher").add("teacher", teacher)
+  ];
+});
+```
+
+In summary:
+
+- Add a value: `person.add("salary", 10)` <-> `person.salary += 10`
+- Remove a value: `person.remove("salary, 10)` <-> `person.salary -= 10`
+- Set a value: `person.remove("salary").add("salary", 10)` <-> `person.salary := 10`
+- Remove an attribute `person.remove("salary")` <-> `person.salary := none`
+- Remove a record `person.remove()` <-> `person := none`
+
```

Comment:
Feels a bit weird to have this summary be in the middle of the guide. I'd probably put it at the top as a TL;DR or the bottom as a summary. Either way I'd have one line for *everything*, including expressions, aggregates, chooses, and unions.